### PR TITLE
Remove obsolete gwt-HTML wrapping from <th> elements.

### LINKED ISSUE #252: Plain string in light DOM thead prevents sorting
Test case: http://jsfiddle.net/8v274qyy/

Plain string content in light DOM `thead th` creates a wrapping GWT HTML widget which eats the events and prevents sorting.

### PR DESCRIPTION
**Pull request by Saulis**
Fixes #252

[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/256)

### CODE REVIEW

**Comment by tomivirkki** [comment]
<sup>**[java/src/main/java/com/vaadin/elements/grid/table/GridLightDomTable.java, line 229 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K2vWW7b6PJ7PF5fvbZ0)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/16fcaafe17875c28c01e52e4d5df852da8ab9e2f/java/src/main/java/com/vaadin/elements/grid/table/GridLightDomTable.java#L229)):</sup>
This was originally wrapped in 2f6266e . The goal was to for example enable focusing an input element placed in the header trough light DOM. Does this still work after this change?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)

**Comment by Saulis** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[java/src/main/java/com/vaadin/elements/grid/table/GridLightDomTable.java, line 229 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K2vWW7b6PJ7PF5fvbZ0)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/16fcaafe17875c28c01e52e4d5df852da8ab9e2f/java/src/main/java/com/vaadin/elements/grid/table/GridLightDomTable.java#L229)):</sup>
Indeed not, unless focused using tab. Is it recreated because of sorting clears the cache?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)

**Comment by Saulis** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[java/src/main/java/com/vaadin/elements/grid/table/GridLightDomTable.java, line 229 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K2vWW7b6PJ7PF5fvbZ0)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/16fcaafe17875c28c01e52e4d5df852da8ab9e2f/java/src/main/java/com/vaadin/elements/grid/table/GridLightDomTable.java#L229)):</sup>
Mm except that it's recreated even when not sortable just by clicking on the header?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)

**Comment by tomivirkki** [comment]
<sup>**[java/src/main/java/com/vaadin/elements/grid/table/GridLightDomTable.java, line 229 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K2vWW7b6PJ7PF5fvbZ0)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/16fcaafe17875c28c01e52e4d5df852da8ab9e2f/java/src/main/java/com/vaadin/elements/grid/table/GridLightDomTable.java#L229)):</sup>
Yep, that's Grid's doings. So we'll need to wrap iff the text contains an element. Same check can be used as in GridColumn#49


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)

**Comment by manolo** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[test/grid-rendering-light-dom.html, line 210 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3-3fEENeqw-HLuK7t6)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/16fcaafe17875c28c01e52e4d5df852da8ab9e2f/test/grid-rendering-light-dom.html#L210)):</sup>
To me this test does not guarantees anything related with the issue.
I would have some use cases: `<th>Foo</th>` `<th><span>Foo<span></th>` `<th><input>Foo<input></th>` content and firing a click event over the text content. And in the case of the input checking that  ie: `onkeyup` events works


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)

**Comment by Saulis** [comment]
Review status: 1 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[test/grid-rendering-light-dom.html, line 210 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3-3fEENeqw-HLuK7t6)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/16fcaafe17875c28c01e52e4d5df852da8ab9e2f/test/grid-rendering-light-dom.html#L210)):</sup>
Rewrote the whole patch, please have a look.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)

File: vaadin-grid-doc.html [hunk 1]
```diff
@@ -423,6 +423,7 @@
 
       /**
        * Contents of the cell.
+       * Note: Assigning an Element object as content disables manual sorting for that cell.
```
Comment (jounik) [inline vaadin-grid-doc.html]: I would rephrase that to “Assigning an HTML element reference as the content for the default header row cell disables end users sorting for that column.” to be absolutely precise what it does.

**Comment by tomivirkki** [comment]
Review status: 1 of 6 files reviewed at latest revision, 5 unresolved discussions.

---

<sup>**[java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java, line 131 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3K4C_eUWvWK_ru2z1m)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/e74d6c2a0d1c88b41176557554e68349c510b052/java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java#L131)):</sup>
if (!isEventTargetFocusedAndInsideStaticSection(event)) {
    super.onBrowserEvent(event);
}

...is cleaner


---

<sup>**[java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java, line 137 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3K4zjViIyNCni8kPfy)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/e74d6c2a0d1c88b41176557554e68349c510b052/java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java#L137)):</sup>
The api is strange if you're only using the event to get the target from inside. Maybe isElementFocusedAndInside...(Element element) which is called with isElement....(event.getEventTarget().cast())


---

<sup>**[java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java, line 138 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3K4ZLdXKCjMOxm6X58)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/e74d6c2a0d1c88b41176557554e68349c510b052/java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java#L138)):</sup>
Deprecated api, use getEventTarget().cast() instead


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)

**Comment by Saulis** [comment]
Review status: 1 of 6 files reviewed at latest revision, 5 unresolved discussions.

---

_[java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java, line 131 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3K4C_eUWvWK_ru2z1m:-K3KAk4OxlpRN7-d3XIF:1753077659) ([raw file](https://github.com/vaadin/vaadin-grid/blob/e74d6c2a0d1c88b41176557554e68349c510b052/java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java#L131)):_
Have to disagree here, I know you prefer not using `return;`  but I think it's clearer than using a negation - it's needs a few reads before your eyes spot the `!`. I thought about reversing the method name but it would become something like `isElementNotFocusedOrInsideStaticSection` which I found also a bit confusing.

---

_[java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java, line 137 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3K4zjViIyNCni8kPfy:-K3KC3-2yh1FQfG3Llvs:-499146009) ([raw file](https://github.com/vaadin/vaadin-grid/blob/e74d6c2a0d1c88b41176557554e68349c510b052/java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java#L137)):_
Done.

---

_[java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java, line 138 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3K4ZLdXKCjMOxm6X58:-K3KC3tm6l5YXMQvbN6L:-499146009) ([raw file](https://github.com/vaadin/vaadin-grid/blob/e74d6c2a0d1c88b41176557554e68349c510b052/java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java#L138)):_
Done.

---

_[vaadin-grid-doc.html, line 426 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3JZfmirBya2xliniTO-r2-426:-K3K9ALRfWRwvZZSxpwi:-499146009) ([raw file](https://github.com/vaadin/vaadin-grid/blob/e74d6c2a0d1c88b41176557554e68349c510b052/vaadin-grid-doc.html#L426)):_
Done.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)_

**Comment by Saulis** [comment]
Review status: 1 of 6 files reviewed at latest revision, 5 unresolved discussions.

---

_[java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java, line 131 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256#-K3K4C_eUWvWK_ru2z1m:-K3KKYB02GiPuP5Wf121:-499146009) ([raw file](https://github.com/vaadin/vaadin-grid/blob/e74d6c2a0d1c88b41176557554e68349c510b052/java/src/main/java/com/vaadin/elements/grid/ViolatedGrid.java#L131)):_
Done.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)_

**Comment by tomivirkki** [comment]
Reviewed 1 of 5 files at r2, 2 of 4 files at r3.
Review status: 4 of 6 files reviewed at latest revision, 2 unresolved discussions.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/256)_
